src/app/store/reducers/mail.reducers.ts

```diff
@@ -42,11 +42,18 @@ export function reducer(state = initialState, action: MailActions): MailState {
     }
 
     case MailActionTypes.MOVE_MAIL_SUCCESS: {
-      const listOfIDs = action.payload.ids.split(',');
+      const listOfIDs = action.payload.ids.toString().split(',');
       state.mails = state.mails.filter(mail => !listOfIDs.includes(mail.id.toString()));
       return { ...state, inProgress: false };
     }
 
+    case MailActionTypes.UNDO_DELETE_MAIL_SUCCESS: {
+      return {
+        ...state,
+        mails: [...state.mails, action.payload.mail],
```

Comment:
why are you adding the undo mail to list of mails : : ` mails: [...state.mails, action.payload.mail],`
user can be on any page, maybe on inbox or trash.

Reply:
@Zain, you didn't fix this

Reply:
I have fixed it.